liquid/validation_test.go

```diff
@@ -0,0 +1,423 @@
+/*******************************************************************************
+*
+* Copyright 2025 SAP SE
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You should have received a copy of the License along with this
+* program. If not, you may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*
+*******************************************************************************/
+
+package liquid
+
+import (
+	"math/big"
+	"slices"
+	"strings"
+	"testing"
+
+	"github.com/sapcc/go-bits/errext"
+)
+
+// TODO: Can there be resources with HasCapacity=false and HasQuota=False or should this be forbidden?
```

Comment:
Yes, this is a valid usecase, e.g. when we only want to report usage of a thing to billing.